stan/math/rev/core/scoped_chainablestack.hpp

```diff
@@ -0,0 +1,50 @@
+#ifndef STAN_MATH_REV_CORE_SCOPED_CHAINABLESTACK_HPP
+#define STAN_MATH_REV_CORE_SCOPED_CHAINABLESTACK_HPP
+
+#include <stan/math/rev/core/chainablestack.hpp>
+#include <stan/math/rev/core/start_nested.hpp>
+#include <stan/math/rev/core/recover_memory_nested.hpp>
+
+#include <vector>
+
+namespace stan {
+namespace math {
+
+class ScopedChainableStack {
+  typedef ChainableStack::AutodiffStackStorage chainablestack_t;
+  chainablestack_t local_stack_;
+
+  using stack_queue_t = std::vector<chainablestack_t*>;
+  stack_queue_t stack_queue_;
```

Comment:
[optional] if we only use these here I think not using an alias is better

```suggestion
  ChainableStack::AutodiffStackStorage local_stack_;

 std::vector<ChainableStack::AutodiffStackStorage*> stack_queue_;
```